### ISSUE TITLE
bibtex fails on MacOS

### ISSUE THREAD

**Issue by mneuhaus-cv** [doxygen/doxygen#7525]
Any call to bibtex fails with the error:

```
Generating citations page...
error opening bib1654.bbl: No such file or directory
bibtex command failed: No such file or directory
error: Problems running bibtex. Verify that the command 'perl --version' works from the command line. Exit code: 2
```

The bibtex input file (bib1654.aux) **does exist** in the output directory
Both bibtex (MacTeX) and Perl (Xcode cmd line tools) **are** installed:

```
MBP-MNS:~ mneuhaus$ bibtex --version
BibTeX 0.99d (TeX Live 2019)
kpathsea version 6.3.1
Copyright 2019 Oren Patashnik.
...

MBP-MNS:~ mneuhaus$ perl --version

This is perl 5, version 18, subversion 4 (v5.18.4) built for darwin-thread-multi-2level
(with 2 registered patches, see perl -V for more detail)

Copyright 1987-2013, Larry Wall
...
```

Same project works just fine on Windows (10, TeX Live 2017).

Tested with Doxygen 1.8.15 ... 1.8.17.
I'm at a total loss here.

[bibtexTest.zip](https://github.com/doxygen/doxygen/files/4091357/bibtexTest.zip)

**Comment by albert-github** [comment]
I don't have a Mac so I cannot directly test this.
There has been a problem with MiKTeX (this is for Windows !) where the bibtex executable was missing (https://github.com/MiKTeX/miktex/issues/443) due to the fact that it is part of miktex-misc that was not declared essential anymore. It might be that this has a similar background.
Can you check:
- where bibtex executable resides (it should be at the same directory as pdflatex).

**Comment by mneuhaus-cv** [comment]
Both reside in the same directory (/Library/TeX/texbin);
and for that matter: the directory **is** part of the $PATH

(saw your search for perl ticket)

**Comment by albert-github** [comment]
Strange problem (thanks for the zip file)
The perl version looks a bit old, but I don't think this should cause a problem (which version is used on Windows ?), before diving into the perl script itself:
- Can you run the following test with `doxygen -d extcmd`: [tst.zip](https://github.com/doxygen/doxygen/files/4091752/tst.zip)
- In the `bib` file I do see quite a few accented characters like "ü" etc., can you run a test where there are in the `bib` files no accented characters?

**Comment by mneuhaus-cv** [comment]
showed the expected (??) result:  [tst.zip](https://github.com/doxygen/doxygen/files/4095852/tst.zip)
```
sh: latex: command not found
error: Problems running latex. Check your installation or look for typos in _formulas.tex and check _formulas.log!
Generating image form_0.png for formula
sh: dvips: command not found
```
OK, so there seems indeed to be a problem with $PATH

**Comment by albert-github** [comment]
This was indeed the intention of this test, to see whether or not the executables (and its "friends) could be found.
Might it be that you normally use another shell than `sh` (e.g. `zsh` or `bash`) and that this shell has a different path than the `sh`?

I think this is a local installation issue, setting this issue to usage (though still interested in the solution).

**Comment by mneuhaus-cv** [comment]
yep, I'm indeed using a diferent "shell" - Eclipse (or Doxygen.app - the graphical I/F).

And those application bundles (xxx.app) DON'T inherit the $PATH, when started directly.
To circumvent this, one can start the executable (Eclipse.app/Contents/MacOS/eclipse (using "Show Package Contents")) directly; this creates a shell and starts eclipse from there...

Found this via a detour using Homwbrew to install cmdline doxygen, which works just fine btw.
https://superuser.com/questions/28344/path-env-variable-on-mac-os-x-and-or-eclipse was helpful as well.

Long story short: bibtex / latex work fine, when $PATH is set - and evaluated - correctly.

Given that your Doxygen.app GUI doesn't work with bibtex under MacOS either (the examples I uploaded **were** created with the Doxygen GUI!), you may consider to porbably include some means of adding to the search $PATH in the Doxyfile?

**Comment by albert-github** [comment]
I'm not sure of adjusting the `PATH` inside doxygen, I think the problem lies a bit on the other side that the system has not be installed properly (a bit a blunt statement), by setting the right `PATH` in the `.profile` (when I'm not mistaken) the problem can be solved as well.
@doxygen ?

**Comment by mneuhaus-cv** [comment]
coming back to your "a bit a blunt statement"...

It's Apples intention to *not* inherit the shell $PATH for application bundles.
App bundles *shall* work without the need for external tools, therefor such extension is not required (Apples interpretation). Adding to the $PATH of app bundles was restricted more and more by Apple over time (you'll find loads of how-tos to "fix" this).

**Comment by albert-github** [comment]
@mneuhaus-cv Which actions should doxygen  undertake to make it working under MacOS in your opinion?

**Comment by mneuhaus-cv** [comment]
on second thought ... (mayby third thought)

you're right, that this shouldn't be a setting in the doxyfile (it couldn't be used in a different installation)

As mentioned above, this "problem" exists in the Doxygen GUI as well;
so maybe include a setting there to configure the latex/bibtex paths.

For Eclipse integration this should be fixed in eclox (the Eclipse Doxygen plugin).

Thanks for your time anyway!

**Comment by albert-github** [comment]
The doxygen GUI (doxywizard) uses the doxygen executable to perform the documentation generation and thus a Doxyfile. A setting in the doxygen GUI automatically leads to a setting in the Doxyfile and vice versa.
For a number of external programs doxygen can use explicit paths (e.g. `DOT_PATH`, and when not set falls back to the `PATH` as set in the environment).
